pkg/api/controller.go

```diff
@@ -3025,39 +3062,41 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
-func (c *Controller) sendResetPasswordEmail(email string, token string) error {
-	return c.Emailer.SendEmailWithLimit([]string{email}, token, token, nil)
-}
-
-func (c *Controller) ForgotPassword(w http.ResponseWriter, r *http.Request, body ForgotPasswordJSONRequestBody) {
-	addr, err := mail.ParseAddress(body.Email)
+func (c *Controller) resetPasswordRequest(ctx context.Context, email string) error {
+	addr, err := mail.ParseAddress(email)
 	if err != nil {
-		c.Logger.WithError(err).WithField("email", body.Email).Debug("forgot password with invalid email")
-		w.WriteHeader(http.StatusNoContent)
-		return
+		c.Logger.WithError(err).WithField("email", email).Debug("reset password with invalid email")
+		return err
 	}
-	user, err := c.Auth.GetUserByEmail(r.Context(), addr.Address)
+	user, err := c.Auth.GetUserByEmail(ctx, addr.Address)
 	if err != nil {
 		c.Logger.WithError(err).WithField("email", addr.Address).Debug("failed to retrieve user by email")
-		w.WriteHeader(http.StatusNoContent)
-		return
+		return err
 	}
-	email := StringValue(user.Email)
+	email = StringValue(user.Email)
 	secret := c.Auth.SecretStore().SharedSecret()
 	currentTime := time.Now()
 	token, err := GenerateJWTResetPassword(secret, email, currentTime, currentTime.Add(DefaultResetPasswordExpiration))
 	if err != nil {
 		c.Logger.WithError(err).WithField("email", email).Debug("failed to create a token")
-		writeError(w, http.StatusInternalServerError, err)
-		return
+		return err
 	}
-	// TODO (@shimi9276) create template for sending the email with link for reset
-	err = c.sendResetPasswordEmail(email, token)
+	// TODO (@shimi9276) create template for sending the email with link for reset using the passed template
+	err = c.Emailer.SendEmailWithLimit([]string{email}, token, token, nil)
 	if err != nil {
 		c.Logger.WithError(err).WithField("email", email).Warn("failed sending reset password email")
+		fmt.Println(err)
 	} else {
 		c.Logger.WithField("email", email).Info("reset password email sent")
 	}
+	return nil
+}
+
+func (c *Controller) ForgotPassword(w http.ResponseWriter, r *http.Request, body ForgotPasswordJSONRequestBody) {
+	err := c.resetPasswordRequest(r.Context(), body.Email)
+	if err != nil {
+		return
```

Comment:
Need to handle write error response

Reply:
Fixed as the calling function will do that

Reply:
Please check the code - the function returns an err - so it write an error as a response or return one - for each case something is missing here. No content is also a response.

Reply:
your right, fixed